File: -Errors-/P1024/P1024.py
test_cases = int(input())

for i in range(test_cases):
    text = input()
    new_text = ''

    for l in text:
        if l.isalpha():
            new_text += chr(ord(l) + 3)
        else:
            new_text += l

    new_text = new_text[::-1]
    half = int((len(new_text) / 2))
    first_part = new_text[0:half]
    second_part = new_text[half:]
    third_part = ''

    for l in second_part:
        third_part += chr(ord(l) - 1)

    encrypted_text = first_part + third_part

    print(encrypted_text)

File: Strings/P1024/src/P1024.kt
fun main(args: Array<String>) {

    val testCases = readLine()?.toInt()!!

    for (i in 1..testCases) {

        val text = readLine().toString()
        var newText = ""

        for (j in text) {
            newText += if (j.isLetter()) {
                (j.toByte().toInt() + 3).toChar()
            } else {
                j
            }
        }

        newText = newText.reversed()
        val half = (newText.length / 2)
        val firstPart = newText.substring(0, half)
        val secondPart = newText.substring(half, newText.length)
        var thirdPart = ""

        for (k in secondPart) {
            thirdPart += (k.toByte().toInt() - 1).toChar()
        }

        val encryptedText = firstPart + thirdPart
        print("$encryptedText\n")
    }
}